packages/metal-state/src/validators.js

```diff
@@ -61,6 +63,10 @@ const validators = {
 	 * @return {!function()}
 	 */
 	objectOf: function(validator) {
+		const validatorResult = validators.func(validator);
```

Comment:
Hey @diegonvs 

I noticed that in `arrayOf` and `objectOf` you validate the args outside the `maybe` function, but in `shapeOf` it's done inside the `maybe` function. This will result in two different behaviors.

With how it is currently, `validators.arrayOf(2)` will return an error, but `validators.shapeOf(2)` will not.

I'm pretty sure we want the behavior of `validators.shapeOf`, since returning an error immediately when using the `Config` helpers won't actually print the error. For example,

```javascript
const config = Config.arrayOf(2); // TypeError: Cannot read property 'validator' of undefined
```

Reply:
Hey @Robert-Frampton, 

One thing confusing about that validation in particular is that it's a different scenario than the others. Validations inside the maybe will only get thrown once the user of your component is trying to pass an invalid argument to a components state property.

The error thrown by 

```JS
MyComponent.STATE = {
    items: Config.arrayOf(2);
}
```

on the other hand, is more targeted towards component developers instead of component users. And should probably be thrown upon component definition (as soon as possible) instead of waiting for component instantiation.

>I'm pretty sure we want the behavior of validators.shapeOf, since returning an error immediately when using the Config helpers won't actually print the error. 

One alternative would be throwing an error right away instead if returning it.

Reply:
I think throwing the error immediately makes more sense to me. In that case I'm fine with it, though I would like to change the error message that is thrown. The message returned from `validators.func` is confusing when you don't pass the additional `name` and `context` args to it.

Reply:
@diegonvs, do you have time to make those changes? If not I can do it, just trying to get ready for a release later this week.

Reply:
Hey @Robert-Frampton, i'm currently working on [domData get/set functions](https://github.com/metal/metal.js/issues/295) these days. Probably this feature will enter on this week release :)  You can take those changes in validators?